llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -448,11 +449,15 @@ VPBasicBlock::createEmptyBasicBlock(VPTransformState::CFGState &CFG) {
 }
 
 void VPIRBasicBlock::execute(VPTransformState *State) {
-  assert(getHierarchicalSuccessors().empty() &&
-         "VPIRBasicBlock cannot have successors at the moment");
-
+  assert(getHierarchicalSuccessors().size() <= 2);
```

Comment:
```suggestion
  assert(getHierarchicalSuccessors().size() <= 2 &&
           "VPIRBasicBlock can have at most two successors at the moment");
```
can any block have more than 2? ...